include/swift/Parse/ParsedRawSyntaxNode.h

```diff
@@ -62,24 +77,52 @@ class ParsedRawSyntaxNode {
   // MARK: - Constructors
 
   ParsedRawSyntaxNode()
-      : Data(nullptr, DataKind::Null), Range(),
+      : Data(nullptr, DataKind::Null),
         SynKind(uint16_t(syntax::SyntaxKind::Unknown)),
         TokKind(uint16_t(tok::unknown)) {}
 
-  ParsedRawSyntaxNode(OpaqueSyntaxNode Opaque, CharSourceRange Range,
-                      syntax::SyntaxKind SynKind, tok TokKind, DataKind DK,
-                      bool IsMissing)
+#ifdef PARSEDRAWSYNTAXNODE_VERIFY_RANGES
+  ParsedRawSyntaxNode(OpaqueSyntaxNode Opaque, syntax::SyntaxKind SynKind,
+                      tok TokKind, DataKind DK, bool IsMissing,
+                      CharSourceRange Range)
       : Data(Opaque, DK), Range(Range), SynKind(uint16_t(SynKind)),
         TokKind(uint16_t(TokKind)), IsMissing(IsMissing) {
     assert(getKind() == SynKind && "Syntax kind with too large value!");
     assert(getTokenKind() == TokKind && "Token kind with too large value!");
   }
```

Comment:
To reduce the duplicated code, can we:
```suggestion
  ParsedRawSyntaxNode(OpaqueSyntaxNode Opaque, syntax::SyntaxKind SynKind,
                      tok TokKind, DataKind DK, bool IsMissing,
                      CharSourceRange Range)
      : ParsedRawSyntaxNode({Opaque, DK}, SynKind, TokKind, IsMissing, Range) {}
```
ditto for `#else` branch.